go/enclave/enclave.go

```diff
@@ -290,19 +297,9 @@ func (e *enclaveImpl) SubmitTx(tx common.EncryptedTx) (common.EncryptedResponseS
 		return nil, fmt.Errorf("enclave could not respond securely to eth_sendRawTransaction request. Cause: %w", err)
 	}
 
-	// Only the sequencer needs to maintain a transaction mempool. Other node types can return early.
-	if e.config.NodeType != common.Sequencer {
-		return encryptedResult, nil
-	}
-
-	if err = e.mempool.AddMempoolTx(decryptedTx); err != nil {
-		return nil, err
-	}
-
 	if e.config.SpeculativeExecution {
```

Comment:
do you know if speculative execution is ever enabled? This changes behaviour for validator nodes if speculativeExecution is set but don't know if there's any impact (I thought that flag was for POBI and so it's probably deprecated)

Reply:
ah! I literally thought the same thing. But moved it to another PR https://github.com/obscuronet/go-obscuro/pull/1092 :)